oscar_odin/mappings/order.py

```diff
@@ -33,17 +39,82 @@ class SurchargeToResource(OscarBaseMapping):
     to_obj = resources.order.Surcharge
 
 
+class DiscountLineToResource(OscarBaseMapping):
+    """Mapping from an order discount line model to a resource"""
+
+    from_obj = OrderLineDiscountModel
+    to_obj = resources.order.DiscountLine
+
+    @odin.assign_field
+    def line(self):
+        """map line object as resource"""
+        return LineToResource.apply(self.source.line)
+
+
 class DiscountToResource(OscarBaseMapping):
     """Mapping from an order discount model to a resource."""
 
     from_obj = OrderDiscountModel
     to_obj = resources.order.Discount
 
     @odin.map_field
-    def category(self, value: str) -> resources.order.DiscountCategory:
+    def category(self, value: str):
         """Map category."""
         return resources.order.DiscountCategory(value)
 
+    @odin.map_field(from_field="category")
+    def is_basket_discount(self, category) -> bool:
+        """map is basket discount function"""
+        return category == "Basket"
```

Comment:
Can you please return the model property here? So self.source.is_basket_discount

Otherwise if something is changed in the model it will not show so in odin.